src/librustc_mir/transform/match_branches.rs

```diff
@@ -78,7 +78,7 @@ impl<'tcx> MirPass<'tcx> for MatchBranchSimplification {
                         );
                         if let Some(c) = c.literal.try_eval_bool(tcx, param_env) {
                             let op = if c { BinOp::Eq } else { BinOp::Ne };
-                            *rhs = Rvalue::BinaryOp(op, Operand::Move(discr), const_cmp);
+                            *rhs = Rvalue::BinaryOp(op, Operand::Copy(discr), const_cmp);
```

Comment:
Should this mirror the type that is matched in `discr`, rather than always be `Copy`?

Reply:
AFAIU move leaves undefined data behind and moving from it again would be undefined behaviour. At least in theory, if not currently in practice. Since we might need to use it more than once, using copy should be appropriate.

Reply:
Ah makes sense, I wonder if this should always be done if the enum is relatively large